### ISSUE TITLE
Changing Anchors and Strides

### ISSUE THREAD

**Issue by rodrigo2019** [fizyr/keras-retinanet#396]
When I change this [line](https://github.com/fizyr/keras-retinanet/blob/master/keras_retinanet/models/retinanet.py#L191) for values like `[16, 32, 64, 128, 256]` I got better loss results in my training, but the evaluation always return mAP=0.000
What is missing?

**Comment by rsingla92** [comment]
Can you answer the below:

1. What commands are you running to train and to evaluate?
2. Can you report back what is returned at the line below (whichever is relevant):
https://github.com/fizyr/keras-retinanet/blob/master/keras_retinanet/utils/eval.py#L81

or

https://github.com/fizyr/keras-retinanet/blob/master/keras_retinanet/utils/coco_eval.py#L37

Given that your change is in the model itself, and that training is reportedly working, I'd focus on the `evaluate.py` script. That script in turn calls the `eval.py` or `coco_eval.py` found in utils. Skimming over each, I'd hazard a guess that the results from either of those lines are returning something incorrectly. Maybe that's a good starting point for further investigation.

**Comment by hgaiser** [comment]
Hmm interesting, because the line you mentioned doesn't have anything to do with the training of retinanet. In fact, in the latest version, that set of parameters isn't even used in the model while training (only during inference). Are you sure you saw a reduction in loss?

There are 2 places where anchors are generated: one during training and one during testing. Needless to say these anchors should be generated identically, so if you want to experiment with anchors, change them in both places. The one for training is located in the generator, the one for inference you already found.

**Comment by rodrigo2019** [comment]
@rsingla92  and @hgaiser  thank you for your answers, I will make new training to compare and I will post the results here.

**Comment by rodrigo2019** [comment]
@hgaiser I was wrong, the losses are same, but now changing the anchor in the generator I got different results, thank you.
ps: I did the tests before the the NMS improvement
